core/tests/behavior/append.rs

```diff
@@ -0,0 +1,240 @@
+use opendal::Operator;
+
+use super::utils::*;
+
+/// Test services that meet the following capability:
+///
+/// - can_read
+/// - can_write
+/// - can_append
+macro_rules! behavior_append_test {
+    ($service:ident, $($(#[$meta:meta])* $test:ident),*,) => {
+        paste::item! {
+            $(
+                #[test]
+                $(
+                    #[$meta]
+                )*
+                fn [<append_ $test >]() -> anyhow::Result<()> {
+                    match OPERATOR.as_ref() {
+                        Some(op) if op.info().can_read()
+                            && op.info().can_write()
+                            && op.info().can_append() => RUNTIME.block_on($crate::append::$test(op.clone())),
+                        Some(_) => {
+                            log::warn!("service {} doesn't support append, ignored", opendal::Scheme::$service);
+                            Ok(())
+                        },
+                        None => {
+                            Ok(())
+                        }
+                    }
+                }
+            )*
+        }
+    };
+}
+
+#[macro_export]
+macro_rules! behavior_append_tests {
+     ($($service:ident),*) => {
+        $(
+            behavior_append_test!(
+                $service,
+
+                test_append,
+                test_append_to_non_existing_target,
+                test_append_with_dir_path,
+                test_append_with_cache_control,
+                test_append_with_content_type,
+                test_append_with_content_disposition,
+
+                test_fuzz_appender,
+            );
+        )*
+    };
+}
+
+/// Test append to a file must success.
+pub async fn test_append(op: Operator) -> Result<()> {
+    let path = uuid::Uuid::new_v4().to_string();
+    let (content_one, size_one) = gen_bytes();
+    let (content_two, size_two) = gen_bytes();
+
+    op.write(&path, content_one.clone())
+        .await
+        .expect("write file must success");
+
+    op.append(&path, content_two.clone())
+        .await
+        .expect("append file must success");
+
+    let bs = op.read(&path).await.expect("read file must success");
+
+    assert_eq!(bs.len(), size_one + size_two);
+    assert_eq!(bs[..size_one], content_one);
+    assert_eq!(bs[size_one..], content_two);
+
+    op.delete(&path).await.expect("delete file must success");
+
+    Ok(())
+}
+
+/// Test append to a non existing target must success.
+pub async fn test_append_to_non_existing_target(op: Operator) -> Result<()> {
+    let path = uuid::Uuid::new_v4().to_string();
+    let (content, size) = gen_bytes();
+
+    op.append(&path, content.clone())
+        .await
+        .expect_err("append to non existing target must success");
+
+    let bs = op.read(&path).await?;
+    assert_eq!(bs.len(), size);
+    assert_eq!(bs, content);
+
+    Ok(())
+}
+
+/// Test append to a directory path must fail.
+pub async fn test_append_with_dir_path(op: Operator) -> Result<()> {
```

Comment:
We don't need to test this. It doesn't worth.

Reply:
Don't worth it? This is also a behavior.

Reply:
It's a bit like a test of `1==1` to me.

Reply:
I prefer to keep it. 🤔 Since it is indeed part of the append's behavior. We *can't* append to a dir.

Reply:
> `write` also has a test like this.

I do think it's meaningless.

Reply:
> I do think it's meaningless.

I don't mean that it's meaningless. 
It's just that behavior tests should be as comprehensive as possible. 👀